Marlin/src/module/temperature.cpp

```diff
@@ -86,12 +86,12 @@ Temperature thermalManager;
  */
 
 #if HAS_HEATED_BED
-  #define _BED_PSTR(M,E) (E) == -1 ? PSTR(MSG_BED ": " M) :
+  #define _BED_PSTR(M,E) (E) == -1 ? PSTR(M ## _BED) :
```

Comment:
That's the part that doesn't compile. `M` get's expanded all the way to a string and it attempts to append `_BED` to it. I'm not sure if there's a way to make this work with the current setup, other than switching on the message type an selecting the one for the bed/chamber.

Reply:
It can be made to work. I will experiment further.

Reply:
Perhaps moving the underscore to the call site? As long as it doesn't expand any further it should append and then expand correctly, right? So, `PSTR(M ## BED)` here and `MSG_THERMAL_RUNAWAY_` at the call site?

Reply:
A commit has been added that should do the trick.